src/Microsoft.Health.Dicom.Blob/Features/Health/DicomConnectedStoreHealthCheck.cs

```diff
@@ -0,0 +1,102 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using EnsureThat;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Health.Core.Features.Health;
+using Microsoft.Health.Dicom.Blob.Extensions;
+using Microsoft.Health.Dicom.Blob.Features.Storage;
+
+namespace Microsoft.Health.Dicom.Blob.Features.Health;
+
+internal class DicomConnectedStoreHealthCheck : IHealthCheck
+{
+    private readonly string _degradedDescription = "The health of the connected store has degraded.";
+
+    private readonly string _testContent = "Test content.";
+    private readonly string _testDirectoryName = "DicomServiceHealthCheck";
```

Comment:
Seeing this doc https://learn.microsoft.com/en-us/azure/healthcare-apis/dicom/dicom-data-lake#data-contracts is making me rethink the name. 

Should we do AHDSHealthCheck/healthCheck.txt (or something like that?)

Or, if we are going to store all of the dicom files under AHDS anyways, should we put this there as well? AHDS/HealthCheck/healthCheck.txt ?

Reply:
Decided to store under workspace/dicom directory